changelogs/fragments/mass-remove-deprecated.yaml

```diff
@@ -0,0 +1,8 @@
+minor_changes:
+- roles - removed deprecated functionality for non YAML role specs (https://github.com/ansible/ansible/pull/44320)
+- vault - removed deprecated functionality for insecure VaultAES class (https://github.com/ansible/ansible/pull/44320)
+- gather_subset - removed deprecated functionality for using comma separated list with gather_subset (https://github.com/ansible/ansible/pull/44320)
```

Comment:
I might be 'late to the party' but why not allow comma separated strings for  all list types on playbook like we do for modules?

Reply:
Unsure, I just removed what was deprecated.  The deprecation was added at the same time `barelist` was added: https://github.com/ansible/ansible/commit/30a38f94ce81314019fb2d9496602eef0a5b36fe

It's been deprecated since the 2.1 release.